### PR TITLE
Add Petrichor Keyboard to QMK

### PR DESCRIPTION
**Pull request by awkannan**
## Description

Add the petrichor PCB firmware to QMK

The firmware is for both solderable and hotswap
Firmware has been tested on both variants of the PCB

It doesn't adhere to any community layouts as far as I know
All ambiguous keys in LAYOUT_all are covered in other layouts in the info.json. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cannonkeys/petrichor/config.h [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+#pragma once
+
+#define RP2040_BOOTLOADER_DOUBLE_TAP_RESET
+#define RP2040_BOOTLOADER_DOUBLE_TAP_RESET_TIMEOUT 500U
+
+#define PICO_XOSC_STARTUP_DELAY_MULTIPLIER 64
+
+#define RGBLIGHT_LAYERS
+#define RGBLIGHT_DEFAULT_MODE RGBLIGHT_MODE_RAINBOW_SWIRL
+#define RGBLIGHT_LAYERS_OVERRIDE_RGB_OFF
```
Comment (waffle87) [inline keyboards/cannonkeys/petrichor/config.h:13]:
```suggestion
```

File: keyboards/cannonkeys/petrichor/info.json [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+    },
+    "processor": "RP2040",
+    "rgblight": {
+        "animations": {
+            "rainbow_mood": true,
+            "rainbow_swirl": true,
+            "static_gradient": true,
+            "twinkle": true
+        },
+        "led_count": 3
```
Comment (waffle87) [inline keyboards/cannonkeys/petrichor/info.json:28]:
```suggestion
        "led_count": 3,
        "layers": {
            "enabled": true,
            "override_rgb": false
        },
        "default": {
            "animation": "rainbow_swirl"
        }
```

**Comment by awkannan** [comment]
Thank you @waffle87 !

**Review by waffle87 — APPROVED** [review]

**Comment by awkannan** [comment]
@lesshonor Thanks for the feedback! I think since RGBLight is simpler and also already written, I'm not gonna change it now - but this is great for future reference. I'll keep this in mind moving forward.

**Review by drashna — APPROVED** [review]

**Pull request merged**